### PR TITLE
fix: [PL-42341]: Update UserGroupAggregate data "onMemberListRefetch"

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
### Summary

JIRA: https://harness.atlassian.net/browse/PL-42341
Update UserGroupAggregate data "onMemberListRefetch"

On the UserGroup Details page, when a user (say UserA) is removed from the UserGroup & immediately after that another user (say UserB) is added to that UserGroup, UserA gets added back automatically. This was happening because on removal of a user, cached data was not cleaned up properly from UI. This has now been fixed. Now, on first removing UserA, then adding UserB—only UserB shows up as the final addition of this two-step process.

#### Screenshots

BEFORE: (Remove UserA, Add UserB --> leads to eventually both users being added instead of just UserB)

https://github.com/harness/harness-core-ui/assets/96057095/abfae646-8936-44f4-aabe-7b0901b11af1

AFTER: (Remove UserA, Add UserB --> leads to eventually only UserB to be added, as expected)

https://github.com/harness/harness-core-ui/assets/96057095/3fdddf0f-3eb3-4761-b28b-67df8247c89c

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by mayankVermaHarness** [comment]
trigger messagecheck

**Review by namanharness — DISMISSED** [review]

File: src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx [hunk 1]
```diff
@@ -234,6 +239,11 @@ const UserGroupDetails: React.FC = () => {
             managed={userGroup.harnessManaged}
             linkedSSOType={userGroup.linkedSsoType}
             isUserGroupManaged={userGroup.externallyManaged}
+            onMemberListRefetch={(users?: UserMetadataDTO[]) => {
+              if (users) {
+                setUserGroupAggregateResponse({ ...userGroupAggregateResponse, users })
```
Comment (avinashmadhwani02) [inline src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:244]: Data in this component will become inconsistent, "data" variable coming from API will have less Users than userGroupAggregateResponse. 

Shouldn't source of Users be at one place and we should just refetch it from API?

Comment (mayankVermaHarness) [reply src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:244]: Good catch! I wanted the Users to have a single source of truth too.

The problem here is that `<MemberList/>` component needs to render paginated responses—so there are 2 different APIs fetching Users.

Let me see if there's another solution to this.

Comment (mayankVermaHarness) [reply src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx]: Refetching "useGetUserGroupAggregate" to get latest users list seems like the only option here. I have made this change.

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by namanharness — APPROVED** [review]

**Pull request merged**